cns/restserver/ipam.go

```diff
@@ -116,6 +116,23 @@ func (service *HTTPRestService) RequestIPConfigHandler(w http.ResponseWriter, r
 		return
 	}
 
+	// Check the status of the NC
+	ncStatuses := multiContainerStatus(service.state.ContainerStatus)
+	if unsuccessfulStatuses := ncStatuses.GetUnsuccessfulStatusErrors(); unsuccessfulStatuses != "" {
+		// If the status is anything other than success, we send a response back with the actual status and NC ID.
+		reserveResp := &cns.IPConfigResponse{
+			Response: cns.Response{
+				ReturnCode: types.FailedToAllocateIPConfig,
+				Message:    unsuccessfulStatuses,
```

Comment:
All you are trying to do is get the NC Status and create an error message from it. So change the name of the method to 'GetNcStatusErrorMessage'